src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java

```diff
@@ -53,6 +59,27 @@ public boolean isApplicable(Class<? extends Job> jobType) {
     @Extension
     public final static class CompressBuildlogRunListener extends RunListener<Run> {
 
+        private static boolean hasBuildCompressorConfigured(Run<?, ?> run) {
+            Job<?, ?> parent = run.getParent();
+            
+            // Check project for BuildCompressor configuration
+            if (parent.getProperty(BuildLogCompressor.class) != null) {
+                return true;
+            }
+            
+            // Multi-configuration runs have an extra parent to get to the project. Check that too.
+            if (parent.getParent() instanceof Job) {
```

Comment:
Let me know if this looks safe enough. The `instanceof` guard should be sufficient to prevent any NPE on lines 72-73, but I don't know much about the classes I'm querying into.